licences-specs/src/test/groovy/uk/gov/justice/digital/hmpps/licences/specs/postDecision/CaTaskListSpec.groovy

```diff
@@ -165,4 +170,59 @@ class CaTaskListSpec extends GebReportingSpec {
     then: 'I see the BASS offer page'
     at BassOfferPage
   }
+
+
+
+  def 'Shows correct button labels for tasks when HDC REFUSED'() {
+    given: 'An refused licence'
+    testData.loadLicence('decision/refused')
+
+    when: 'I view the task list'
+    to TaskListPage, testData.markAndrewsBookingId
+
+    then: 'I see the task buttons and the submit button'
+    taskListActions.size() == 9
+
+    and: 'All editable tasks have View/Edit buttons'
+    taskListActions.take(7).every { it.text() == 'View/Edit' }
+
+    and: 'Final checks task is view only'
+    taskListAction(tasks.finalChecks).text() == 'Change'
+
+    and: 'Resubmit to Dm is available'
+    taskListAction(tasks.resubmit).text() == 'Resubmit'
+
+    and: 'Postpone/Refuse button is not displayed'
+    !postpone.displayed
+
+    and: 'Create Licence is not displayed'
+    !createLicence.displayed
+
+
+  }
+
+  @Unroll
+  def '#task button links to page'() {
```

Comment:
this seems like a copy of an above test with an extra check for resubmit?
could you not just add the check to the test above?

These tests take a long time so we want to try and minimise the number of tests in this level

Reply:
I did consider that, but the tests are slightly different because the tasklist items are different for Approved compared to Rejected cases.  
In the Rejected case, the task to print the licence is not displayed. 

tasks.create      | LicenceTemplatePage